tests/unit/components/VsCodeExtension.spec.js

```diff
@@ -0,0 +1,31 @@
+import { mount } from '@vue/test-utils';
+import VsCodeExtension from '@/pages/VsCodeExtension.vue';
+import data from '../fixtures/user_page_scenario.appmap.json';
+
+describe('VsCodeExtension.vue', () => {
+  let wrapper;
+
+  beforeEach(() => {
+    wrapper = mount(VsCodeExtension, {
+      stubs: {
+        VDiagramComponent: true,
+        VDiagramTrace: true,
+      },
+    });
+    wrapper.vm.loadData(data);
+  });
+
+  it('sets the selected object by FQID', () => {
+    // FQID: Header
+    const expected = {
+      'label:json': 'json',
+      'event:44': 'User.find_by_id!',
+      'class:app/models/User': 'User',
+    };
+
+    Object.entries(expected).forEach(([fqid, header]) => {
+      wrapper.vm.setSelectedObject(fqid);
+      expect(wrapper.get('details-label__header-title').text()).toMatch(header);
+    });
```

Comment:
We need to make sure we're testing the method and it's working as expected. There's a couple of issues with this test, however. First, we're exporting `__esModule` somewhere in the dependency tree (`Trace.vue`). This causes an issue in Jest, but it's also necessary. We'll need a workaround.

Second, with the `__esModule` export removed, this test still raises an `Invalid Component definition` error. I'm not sure why.

Reply:
See the comment here in Trace.vue for the `__esModule` issue:
https://github.com/applandinc/appmap-js/blob/main/src/components/trace/Trace.vue#L54-L56

Essentially we're trying to hit this block, but IIRC the ESM bundling causes it to fail: 
https://github.com/vuejs/vue/blob/b51430f598b354ed60851bb62885539bd25de3d8/src/core/vdom/helpers/resolve-async-component.js#L20

The explicit export resolves the issue.